aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/AwsPresigningDecorator.kt

```diff
@@ -261,12 +296,17 @@ interface PresignModelTransform {
  * to respect HTTP traits or synthetic equivalents if this is needed for AwsQuery, Ec2Query, or AwsJson.
  */
 class OverrideHttpMethodTransform(
-    private val httpMethodOverrides: Map<ShapeId, String>,
+    httpMethodOverrides: Map<ShapeId, String>,
 ) : PresignModelTransform {
+    private val overrides = httpMethodOverrides.mapKeys { entry -> syntheticShapeId(entry.key) }
+
     override fun transform(model: Model): Model {
         return ModelTransformer.create().mapShapes(model) { shape ->
-            if (shape is OperationShape && httpMethodOverrides.containsKey(shape.id)) {
-                val newMethod = httpMethodOverrides.getValue(shape.id)
+            if (shape is OperationShape && overrides.containsKey(shape.id)) {
+                val newMethod = overrides.getValue(shape.id)
+                check(shape.hasTrait(HttpTrait.ID)) {
```

Comment:
nit: we really should be checking the protocol on the service, having an HTTP trait doesn't actually necessarily mean it's used